packages/atlas/src/themesource/atlas_core/web/core/widgets/_scroll-container-react.scss

```diff
@@ -0,0 +1,208 @@
+    // For top and bottom bars it defines their height
+    // For left and right regions it defines their width
+    // it should be always set, regardless of the region is toggleable or not
+    // For toggleable regions it defines the width of a fully open region
+    //
+    // =========================================
+    // === --toggleable-sidebar-width
+    // Expected on: scroll container top level.
+    // This variable is set from inline styling provided by react components on scroll container top level.
+    //
+    // This variable is a copy of --sidebar-size of a toggleable region (there could be only one toggleable).
+    // This is a duplication because we need to be able to use that value in calculation on
+    // other levels of scroll container while --sidebar-size is available on the region level and not higher
+
+    $sidebar-animation-duration: 250ms;
+
+    .mx-page {
+        //height: 100%; // don't let the page expand below available height of the parent
+
+        display: flex;
+        flex-direction: column;
+    }
+
+    .mx-scrollcontainer {
+        flex-shrink: 0;
+        flex-grow: 1;
+        flex-basis: 0;
+
+        &:not(.mx-scrollcontainer-fixed) {
+            // Make scroll container use full width and grow outside of it's parent
+            // this makes it fully scrollable without individual regions having scroll containers
+            min-height: auto;
+        }
+
+        &.mx-scrollcontainer-fixed {
+            // Make scroll container scrollable per region
+            // For this we fix height to stay withing the size of the parent,
+            // so that regions of the scroll container can't stretch the scroll container
+            // even if they are higher than the screen
+            max-height: 100%;
+            min-height: 100%;
+        }
+    }
+
+    .mx-scrollcontainer-horizontal {
+        display: flex;
+        flex-direction: row;
+    }
+
+    .mx-scrollcontainer-vertical {
+        display: flex;
+        flex-direction: column;
+    }
+
+    .mx-scrollcontainer-fixed {
+        // for scroll per region, regions should overflow if they have long content
+        .mx-scrollcontainer-left,
+        .mx-scrollcontainer-right,
+        .mx-scrollcontainer-center,
+        .mx-scrollcontainer-middle {
+            overflow: auto;
+        }
+    }
+
+    // left and right regions should maintain fixed size, no shrinking or expanding
+    .mx-scrollcontainer-left,
+    .mx-scrollcontainer-right,
+    .mx-scrollcontainer-top,
+    .mx-scrollcontainer-bottom {
+        flex-grow: 0;
+        flex-shrink: 0;
+    }
+    // center region should fill available space
+    .mx-scrollcontainer-middle,
+    .mx-scrollcontainer-center {
+        flex-grow: 1;
+    }
+
+    // if there is a nested scroll container we always want
+    // outer scroll container to stay withing available height
+    // and don't allow inner scroll container to grow
+    .mx-scrollcontainer-nested {
```

Comment:
What I noticed, we changed container types. So, maybe let's reflect this change on class names as well?

Reply:
Or we can't because this name is already used in atlas (and probably old dojo container)?

Reply:
Indeed, we follow the same html structure as it is currently in dojo scroll container.